SDK.md

```diff
@@ -44,6 +44,68 @@ Supports CloudEvents spec milestones and ongoing development version.
 HTTP transport support (both structured and binary).
 Widely used programming language version.
 
+### Object model structure guidelines
+
+The CloudEvents specification set consists of
+
+* the base specification that defines the abstract information model and
+  associated rules for a CloudEvent,
+* a set of extensions to the base specification that define use-case
+  specific attributes and associated rules,
+* a set of event format specifications that define how a CloudEvent and  
+  its attributes are encoded in a particular wire transfer format, and
+* a set of transport bindings that define how a wire-encoded CloudEvent is
+  interoperably transported from a publisher or intermediary to a target
+  endpoint using a particular application protocol.
+
+The object model of any API projection of CloudEvents MUST reflect the
+specification architecture with the following elements:
+
+* A generic CloudEvent class/object/structure that provides access to the
+  CloudEvent attributes and enforces data type constraints and all further
+  conformance rules defined in the core CloudEvents specification. Instances
+  of this class/object/structure MUST always be in a conformant state.
+  The generic CloudEvent abstraction SHOULD provide strongly typed access
+  to all well-known attributes, and MUST also provide name-indexed access
+  to any application-specific extension attributes that the event might carry.
+  Name-indexed access MUST include well-known attributes from the core
+  specification and from application-activated specification extensions (see
+  below). The related rules MUST be enforced when setting the attribute 
+  values through the name-indexed access path.
+* A generic abstraction for extensions that allows for one or more active
+  extensions to be added to the CloudEvent abstraction such that the
+  extension-defined attributes are attached to the CloudEvent and
+  the conformance of their values can be validated.
+  Each SDK MUST implement all extensions defined in the CloudEvents
```

Comment:
What does this mean to "implement" an extension? They at least appear in an extension bag (or something), but does this mean you want them to appear as first-class properties too? If they're just in a bag then I'm not sure what an SDK needs to do to satisfy this MUST.

Reply:
Implementing an extension means that all constraints are checked. And for extensions that we have in the repo, the SDKs should have strongly typed object model and check those constraints.

Reply:
I'm interpreting "strongly types object model" as "well defined getters/setters" - is that correct? If so, if we require all SDKs to implement all repo-defined extensions (e.g. `datadef`) are we then not elevating them to the same level as OPTIONAL spec-defined attributes? Our bar for extensions is pretty low and forcing all SDKs to support them sounds like we should take them more seriously than just "experimental" attributes.